pilot/pkg/bootstrap/server.go

```diff
@@ -426,43 +432,90 @@ func (c *mockController) AppendInstanceHandler(f func(*model.ServiceInstance, mo
 
 func (c *mockController) Run(<-chan struct{}) {}
 
-// initConfigController creates the config controller in the pilotConfig.
-func (s *Server) initConfigController(args *PilotArgs) error {
-	mcpServerAddrs := args.MCPServerAddrs
+func (s *Server) initMCPConfigController(args *PilotArgs) error {
+	clientNodeID := ""
+	supportedTypes := make([]string, len(model.IstioConfigTypes))
+	for i, model := range model.IstioConfigTypes {
+		supportedTypes[i] = fmt.Sprintf("type.googleapis.com/%s", model.MessageName)
+	}
 
-	if len(mcpServerAddrs) > 0 {
-		clientNodeID := args.Namespace
-		supportedTypes := make([]string, len(model.IstioConfigTypes))
-		for i, model := range model.IstioConfigTypes {
-			supportedTypes[i] = fmt.Sprintf("type.googleapis.com/%s", model.MessageName)
-		}
+	options := coredatamodel.Options{
+		DomainSuffix: args.Config.ControllerOptions.DomainSuffix,
+	}
+	mcpController := coredatamodel.NewController(options)
 
-		mcpController := coredatamodel.NewController()
+	ctx, cancel := context.WithCancel(context.Background())
+	var clients []*mcpclient.Client
 
-		s.addStartFunc(func(stop <-chan struct{}) error {
-			ctx, cancel := context.WithCancel(context.Background())
-			go func() {
-				<-stop
-				cancel()
-			}()
-
-			for _, addr := range mcpServerAddrs {
-				// TODO: make this a secure connection before shipping
-				conn, err := grpc.Dial(addr, grpc.WithInsecure())
-				if err != nil {
-					log.Errorf("Unable connecting to MCP Server: %v\n", err)
-					return err
+	for _, addr := range args.MCPServerAddrs {
+		u, err := url.Parse(addr)
+		if err != nil {
+			return err
+		}
+
+		securityOption := grpc.WithInsecure()
+		if u.Scheme == "mcp" {
```

Comment:
Since we are doing url.Parse here, should we change the description for mcpServerAddrs to indicate that they should include mcp://?